src/protocol/codecs/payload.rs

```diff
@@ -6,15 +6,18 @@ use tracing::Span;
 
 use crate::protocol::{
     codecs::{
+        msgpack::ProposalPayload,
         tagmsg::Tag,
         topic::{MsgOfInterest, TopicCodec},
     },
     invalid_data,
 };
 
 #[derive(Debug)]
+#[allow(clippy::enum_variant_names)]
```

Comment:
Is this clippy allow necessary here, the variants don't share the same prefixes or suffixes?

Reply:
Or does it detect `ProposalPayload`?

Reply:
> Or does it detect ProposalPayload?

This is the answer: an enum element contains the name of the enum...
And I wanted to keep the full name of the message since that's the name of that message in the go-algorand repo.